### PR TITLE
Fix Callback Routing

### PR DESCRIPTION
**Pull request by IdanCo**
Function 'removed' in TreeNodeController referred to parent scope, but callback was set in the same scope. Deleted redundent function from Tree**Node**Controller, and updated the function in Tree**Nodes**Controller to trigger callback.

### CODE REVIEW

**Comment by IdanCo** [comment]
btw, I recommend doing gulp test before merging, I couldn't do it from my office computer...

File: dist/angular-ui-tree.js [hunk 1]
```diff
@@ -456,7 +450,7 @@
             // and the 'max-depth' attribute in `ui-tree` or `ui-tree-nodes`.
             // the method can be overrided
             callbacks.accept = function (sourceNodeScope, destNodesScope, destIndex) {
-              return !(destNodesScope.nodropEnabled || destNodesScope.outOfDepth(sourceNodeScope));
+              return !(destNodesScope.nodropEnabled || destNodesScope.$treeScope.nodropEnabled || destNodesScope.outOfDepth(sourceNodeScope));
```
Comment (IdanCo) [inline dist/angular-ui-tree.js]: This isn't mine!
Probably an earlier fix that some how slipped 'gulp build'

File: dist/angular-ui-tree.js [hunk 2]
```diff
@@ -1092,7 +1086,9 @@
               };
             }
 
-            scope.$watch(attrs.maxDepth, function (val) {
+            scope.$watch(function () {
```
Comment (IdanCo) [inline dist/angular-ui-tree.js]: Same here

File: dist/angular-ui-tree.js [hunk 3]
```diff
@@ -1287,7 +1283,7 @@
 
                 // cloneEnabled and cross-tree so copy and do not remove from source
                 if (this.isClone() && this.isForeign()) {
-                  this.parent.insertNode(this.index, angular.copy(nodeData));
+                  this.parent.insertNode(this.index, this.sourceInfo.cloneModel);
```
Comment (IdanCo) [inline dist/angular-ui-tree.js]: And here

**Comment by Aaronm14** [comment]
There's a little conversation going on about this in the issues here: https://github.com/angular-ui-tree/angular-ui-tree/issues/586

Thanks for fixing, hopefully it will get merged here soon.

**Comment by Voles** [comment]
@IdanCo thanks for your work! I've reviewed your PR and it's looks ok to merge. Do you have some time to bring your branch up to date with master? Thanks!

File: dist/angular-ui-tree.js (hunk 1 above)
Comment (Voles) [inline dist/angular-ui-tree.js]: Maybe you've merged master into this branch, or rebased onto master?

**Comment by Voles** [comment]
@IdanCo any updates on this? :-)

**Comment by IdanCo** [comment]
On it!

‏בתאריך יום רביעי, 6 בינואר 2016, Niels Dequeker <notifications@github.com>
כתב:

> @IdanCo <https://github.com/IdanCo> any updates on this? :-)
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/angular-ui-tree/angular-ui-tree/pull/673#issuecomment-169458806>
> .
>


-- 
ic

**Comment by IdanCo** [comment]
Done!

**Comment by Voles** [comment]
Thank you very much, @IdanCo! :cake:

**Comment by willbastian** [comment]
Any thought on the issue documented with the removed callback being hit when re-ordering nodes?

**Pull request merged**